### PR TITLE
Update CODEOWNERS

### PR DESCRIPTION
**Pull request by jeredding**
Much like https://github.com/elastic/cloud-sdk-go/pull/385, There's been a formal agreement to move the ecctl over to the applications team and they are actively working on it and unfortunately it requires Delivery to approve PRs. With this change, Delivery will no longer be a blocker.  This shifts the codeowners to the responsible team. 

@pcsanwald @StaceyKingPoling heads up for awareness.

### CODE REVIEW

**Review by rhass — APPROVED** [review]

**Review by StaceyKingPoling — APPROVED** [review]

**Review by artemnikitin — APPROVED** [review]

**Comment by pcsanwald** [comment]
@jeredding I'm -1 on shifting CODEOWNERS to another team that hasn't accepted responsibility yet. It's actually much better if _the accepting team_ does this change. I don't think any team should be moving CODEOWNERS to another team without explicit agreement on that; I'd want to see a PR approval, on this PR, from the accepting team.

If we've done this in other places, I'd ask we revert that.

**Pull request closed without merging**
